rust/src/dcerpc/dcerpc.rs

```diff
@@ -810,10 +880,49 @@ impl DCERPCState {
     pub fn handle_input_data(&mut self, input: &[u8], direction: u8) -> AppLayerResult {
         let mut parsed;
         let retval;
-        let input_len = input.len();
+        let mut cur_i = input;
+        let input_len = cur_i.len();
         let mut v: Vec<u8>;
         // Set any query's completion status to false in the beginning
         self.query_completed = false;
+
+        // Skip the record since this means that its in the middle of a known length record
+        if self.ts_gap || self.tc_gap {
+            SCLogDebug!("Trying to catch up after GAP (input {})", cur_i.len());
+            while cur_i.len() > 0 { // min record size
+                match self.search_dcerpc_record(cur_i) {
+                    Ok((_, pg)) => {
+                        SCLogDebug!("DCERPC record found");
+                        let offset = cur_i.len() - pg.len();
+                        if offset == 1 {
+                            cur_i = &cur_i[offset + 2..];
+                            continue; // see if we have another record in our data
```

Comment:
Why continue ? is not the first record the good one to start ?

Reply:
hmm I considered here that the newest record that we can find is the one that should be the one to start after the gap as it might be followed by its corresponding record. But, now that I think it only makes sense if we're looking for a request after the gap. Get the latest request and hope it is followed by its corresponding response.
Would this make sense?

Reply:
I do not DCEPRC specifics, but what would make sens to me is to parse the maximum data, hence get the earliest record we can...